### ISSUE TITLE
StairHandler can't handle nReversals=0

### ISSUE THREAD

**Issue by datalowe** [psychopy/psychopy#3174]
The documentation for the StairHandler class says this about nReversals:

```
            nReversals:
                The minimum number of reversals permitted.
                If `stepSizes` is a list, but the minimum number of
                reversals to perform, `nReversals`, is less than the
                length of this list, PsychoPy will automatically increase
                the minimum number of reversals and emit a warning.
```

However, this is what's actually in the code ([starting at line 163](https://github.com/psychopy/psychopy/blob/master/psychopy/data/staircase.py#L163)):

```
        try:
            self.stepSizes = list(stepSizes)
        except TypeError:
            # stepSizes is not array-like / iterable, i.e., a scalar.
            self.stepSizes = [stepSizes]

        # [...excluding irrelevant code...]

        if nReversals is None:
            self.nReversals = len(self.stepSizes)
        elif len(self.stepSizes) > nReversals:
            msg = ('Increasing number of minimum required reversals to the '
                   'number of step sizes, (%i).' % len(self.stepSizes))
            logging.warn(msg)
            self.nReversals = len(self.stepSizes)
        else:
            self.nReversals = nReversals
```

This code chunk means that the documentation's "If `stepSizes` is a list..." part is made a moot point. Even if a single float/integer is input for the `stepSizes` argument, this is converted to a single-element list by `self.stepSizes = list(stepSizes)`. 

An example: say that someone passes in `StairHandler(..., nReversals=0, stepSizes=0.1, ...)`. What will happen is that the above check notices that the minimum number of required reversals is lower than the length of the list `[0.1]`, throws a warning, and sets self.nReversals to 1. 

The point is that the above makes it entirely impossible to tell PsychoPy that one doesn't want to enforce a minimum number of reversals (ie nReversals=0), instead there's an absolute requirement of at least one reversal. 

Either the documentation should be changed, if users should never be permitted to set `nReversals=0`, or the if-else logic and type conversion need to be changed so that `nReversals=0` becomes valid.

**Comment by peircej** [comment]
I agree there's a dissociation between the docs and the code. I imagine the code in the if statement changed and the docs weren't updated. 

My instinct here is to fix the docs but tell me about the need to change the code instead. I wonder what the use case would be where you would want a staircase that could terminate with zero reversals (where you might not have entered the region of the threshold at all)? We would certainly need to think about the knock-on effects if we did change it

**Comment by datalowe** [comment]
> My instinct here is to fix the docs but tell me about the need to change the code instead. I wonder what the use case would be where you would want a staircase that could terminate with zero reversals (where you might not have entered the region of the threshold at all)? We would certainly need to think about the knock-on effects if we did change it

In my case, I noticed this as I was debugging an implementation of the RDK test. In it, there are three runs/"staircases", with 60 trials each. The number of trials is predetermined/hardcoded, so it's not related to whether a threshold seems to have been reached or not. Maybe this is unusual, but it's the way it was done in the article I based the implementation on. In practice of course, I can set nReversals to 1 and it won't be a problem, since there's no way someone will answer correctly 60 times in a row. When debugging however I reduced the number of trials/run to just 2, to go through them more quickly. First I noticed that more than 2 trials ran, making me think I'd done something wrong. Then when I noticed the warning message I was confused for a while until I looked in the source code and understood what was going on.

So really, other than debugging this particular setup I don't know of any use case.
